src/duckdb.rs

```diff
@@ -251,14 +251,42 @@ impl DuckDBTableProviderFactory {
     }
 
     pub async fn get_or_init_memory_instance(&self) -> Result<DuckDbConnectionPool> {
+        self.get_or_init_memory_instance_with_size_config(None, None)
+            .await
+    }
+
+    pub async fn get_or_init_file_instance(
+        &self,
+        db_path: impl Into<Arc<str>>,
+    ) -> Result<DuckDbConnectionPool> {
+        self.get_or_init_file_instance_with_size_config(db_path, None, None)
+            .await
+    }
+
+    pub async fn get_or_init_memory_instance_with_size_config(
```

Comment:
```suggestion
    async fn get_or_init_memory_instance_with_size_config(
```

Reply:
Actually, if we rework it to take the builder directly, then this can be public again, and `get_or_init_file_instance` and `get_or_init_memory_instance` just become convenience methods that initialize the builder and pass to `get_or_init_instance_with_builder`